src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -56,7 +56,10 @@ public MessageMetadata GetMessageMetadata(string messageTypeIdentifier)
 
                 foreach (var item in messages.Values)
                 {
-                    if (item.MessageType.FullName == messageTypeIdentifier)
+                    var messageTypeFullName = GetMessageTypeNameWithoutAssembly(messageTypeIdentifier);
+                    
+                    if (item.MessageType.FullName == messageTypeIdentifier ||
+                        item.MessageType.FullName == messageTypeFullName)
```

Comment:
Is there a danger that we match message type without assembly too greedily? Where two types are identical except assembly name and we match the wrong one before we get to the right one? 

I guess once it's cached that's going to happen anyway.

Reply:
@mikeminutillo I added an extra logging to log what we're actually mapping to, if the type was not found.